instance/api/instance.py

```diff
@@ -70,6 +70,14 @@ def get_view_name(self):
         Get the verbose name for each view
         """
         suffix = self.suffix
-        if self.action == 'retrieve':
+        # DRF will only set the `.action` attribute on a view in the context of a request.
+        # In recent versions of DRF, the `get_breadcrumbs` function instantiates a view
+        # and calls `get_view_name` on it outside of the context of a request,
+        # cf. https://github.com/tomchristie/django-rest-framework/pull/3273.
+        # In that case, `.action` is not set on the view, so referencing it directly
+        # (via `self.action`) inside this method will fail with an `AttributeError`.
+        # We guard against that here by using `getattr` to retrieve the value
+        # of the `action` field.
+        if getattr(self, 'action', None) == 'retrieve':
             suffix = "Details"
         return "Instance {}".format(suffix)
```

Comment:
To understand this better, why do we change the view name depending on the action? Is there a broader fix/refactoring which would allow to prevent burying ourselves further here?